lib/dialects/sqlite/query.js

```diff
@@ -24,11 +24,61 @@ Query.prototype.getInsertIdField = function() {
   return 'lastID';
 };
 
-Query.prototype.run = function(sql) {
+/**
+ * rewrite query with parameters
+ */
+Query.prototype.formatBindParameters = function(sql, values, timeZone, dialect) {
+  var bindParam = [];
+  if (Array.isArray(values)) {
+    bindParam = {};
+    values.forEach(function(v, i) {
+      bindParam['$'+(i+1)] = v;
+    });
+    sql = sql.replace(/\$(\$|(?:\d+))/g, function(value, key) {
+      if ('$' === key) {
+        return key;
+      }
+  
+      key = key - 1;
+      if (values[key] !== undefined) {
+        return value;
+      } else {
+        throw new Error('Named bind parameter "' + value + '" has no value in the given object.');
+      }
+    });
+  } else {
+    bindParam = {};
+    if (typeof values === 'object') {
+      Object.keys(values).forEach(function(k) {
+        bindParam['$'+k] = values[k];
+      });
+    }
+    sql = sql.replace(/\$(\$|(?!\d)(?:\w+))/g, function(value, key) {
+      if ('$' === key) {
+        return key;
+      }
+  
+      if (values[key] !== undefined) {
+        return value;
+      } else {
+        throw new Error('Named bind parameter "' + value + '" has no value in the given object.');
+      }
+    });
+  }
+  return [sql, bindParam];
+};
```

Comment:
Same here, looks very similar, what are the actual differences, can they be refactored?